distributed/scheduler.py

```diff
@@ -2548,12 +2563,14 @@ def transition_waiting_released(self, key: Key, stimulus_id: str) -> RecsMsgs:
             assert not ts.who_has
             assert not ts.processing_on
 
-        for dts in ts.dependencies:
-            if ts in dts.waiters:
-                dts.waiters.discard(ts)
+        for dts in ts.dependencies or ():
+            if ts in (dts.waiters or ()):
+                if dts.waiters:
+                    dts.waiters.discard(ts)
                 if not dts.waiters and not dts.who_wants:
                     recommendations[dts.key] = "released"
-        ts.waiting_on.clear()
+        if ts.waiting_on:
+            ts.waiting_on.clear()
```

Comment:
Instead of clearing we can probably reset it to `None`

Reply:
Good point https://github.com/dask/distributed/pull/8331/commits/62bf269536e9431c8097e4a5e99df124596a86db